### PR TITLE
Devops Pipeline to use Release version from Git

### PR DESCRIPTION
**Pull request by sookeke**
- Automate release version number from git instead of manual release version prompt
- Every two weeks before end of every sprint, A release and tag will be created for test and master branch
- The release version number will be used for image tagging and deployment  for consistency
- The image tagging naming convention should follow the semantic [APP_NAME]:[RELEASE_VERSION_NUMBER]-[BRANCH] e.g pims-api:v0.20.7.3-test or pims-app:v02.0.7.3-master

### CODE REVIEW

**Comment by sookeke** [comment]
https://jira.th.gov.bc.ca/browse/PSP-1755

**Comment by devinleighsmith** [comment]
@sookeke please squash your commits, Ideally when a new PR is created it has only 1 commit (although a small number of commits may be appropriate in some cases). Documentation for that is here:
https://github.com/bcgov/PIMS/wiki/GitHub-Workflow#7-squash-your-commits-before-a-pull-request

**Comment by sookeke** [comment]
Yes, go to know. But I can squash on merge. Thank you

File: openshift/4.0/pipelines/Jenkinsfile.cicd [hunk 1]
```diff
@@ -296,7 +225,23 @@ pipeline {
         echo '...'
       }
     }
+    stage('Approve Deployment') {
+      when {
+        beforeInput true
+         expression { 
+             DESTINATION == 'uat' //only ask for approval for UAT and PROD DESTINATION
```
Comment (devinleighsmith) [inline openshift/4.0/pipelines/Jenkinsfile.cicd:232]: comment here says approval is required for UAT and PROD but only checks for uat.

File: tools/keycloak/sync/realm.json [hunk 2]
```diff
@@ -127,7 +127,7 @@
         "AuthorizationServicesEnabled": false,
         "RedirectUris": [
           "https://pims-app-3cd915-prod.apps.silver.devops.gov.bc.ca/api/*",
-          "https://pims.th.gov.bc.ca/api/*"
+          "https:/pims.th.gov.bc.ca/api/*"
```
Comment (devinleighsmith) [inline tools/keycloak/sync/realm.json:130]: what is going on here?

File: tools/keycloak/sync/realm.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
   "Realm": {
     "Name": "72x8v9rw",
-    "DisplayName": "MoTI PIMS Production Environment",
-    "DisplayNameHtml": "<a>MoTI PIMS Production Environment</a>",
+    "DisplayName": "MoTI PIMS PRODUCTION Environment",
```
Comment (devinleighsmith) [inline tools/keycloak/sync/realm.json:4]: Are the changes in this file related to this PR?

File: openshift/4.0/pipelines/Jenkinsfile.cicd (hunk 1 above)
Comment (sookeke) [reply openshift/4.0/pipelines/Jenkinsfile.cicd:232]: @devinleighsmith  PROD has it own seperate pipeline Jenkinsfile.cicd.prod thats why

Comment (sookeke) [reply openshift/4.0/pipelines/Jenkinsfile.cicd:232]: This pipeline us for dev, test and uat

**Comment by devinleighsmith** [comment]
Looks good, I think I'd like to wait until Monday for Alejandro to have a look at this as well.

File: tools/keycloak/sync/realm.json (hunk 3 above)
Comment (sookeke) [reply tools/keycloak/sync/realm.json:4]: NO

**Comment by sookeke** [comment]
> Looks good, I think I'd like to wait until Monday for Alejandro to have a look at this as well.

Yes, wait till monday

File: tools/keycloak/sync/realm.json (hunk 2 above)
Comment (Fosol) [reply tools/keycloak/sync/realm.json:130]: Bad URI

**Comment by devinleighsmith** [comment]
@asanchezr if you could review this one as well.

File: tools/keycloak/sync/realm.json (hunk 2 above)
Comment (sookeke) [reply tools/keycloak/sync/realm.json:130]: fixed

Comment (asanchezr) [reply tools/keycloak/sync/realm.json:130]: @sookeke this line (130) still remains unchanged. Please fix like the others. thanks!

**Comment by asanchezr** [comment]
@sookeke looks good to me - just need one more fix to the keycloak realm URLs (Line 130)

**Review by devinleighsmith — APPROVED** [review]

**Review by asanchezr — APPROVED** [review]

**Review by asanchezr — APPROVED** [review]

**Pull request merged**